plutus-contract/src/Plutus/Contract/StateMachine.hs

```diff
@@ -291,7 +297,9 @@ waitForUpdate ::
     )
     => StateMachineClient state i
     -> Contract w schema e (Maybe (OnChainState state i))
-waitForUpdate client = waitForUpdateTimeout client never >>= awaitPromise >>= \case
+waitForUpdate client = do
+  result <- waitForUpdateTimeout client never >>= awaitPromise
+  case result of
```

Comment:
Unnecessary change?

Reply:
This was for consistency with `waitForUpdateUntilSlot` and `waitForUpdateUntilTime`